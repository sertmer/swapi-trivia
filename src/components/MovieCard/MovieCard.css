.MovieCard {
  border: 1px solid lavender;
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 2%;
  width: 20%;
}

button {
  width: 40%;
}

h2 {
  text-align: center;
}
