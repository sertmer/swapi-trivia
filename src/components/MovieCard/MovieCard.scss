.MovieCard {
  border: 1px solid #f9d71c;
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 2%;
  width: 20%;
  border-radius: 5px;
  height: 40vh;
  justify-content: space-around;
  background-color: #1a1a1a;
}

.characters-button {
  width: 40%;
  margin: 1%;
  border-radius: 5px;
}

h2 {
  text-align: center;
}

h3 {
  font-size: larger;
}

img {
  height: 240px;
  width: 200px;
}
