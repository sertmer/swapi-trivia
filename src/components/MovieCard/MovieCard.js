import React, {Component}from 'react'
import './MovieCard.scss'
import image1 from '../../images/1.jpg'
import image2 from '../../images/2.jpg'
import image3 from '../../images/3.jpg'
import image4 from '../../images/4.jpg'
import image5 from '../../images/5.jpg'
import image6 from '../../images/6.jpeg'
import image7 from '../../images/7.jpg'

const images = [image1, image2, image3, image4, image5, image6, image7]
class MovieCard extends Component {
  constructor(props) {
    super(props) 
    this.state = {
      currentMovie: null
    }
  } 

  findCurrentMovie = (e) => {
    this.setState({ currentMovie: e.target.dataset.category})
    let currentMovie = e.target.dataset.category
    console.log(currentMovie)
    this.props.updateState({ currentMovie })
  }

  render() {
    return (
    <article className='MovieCard'>
      <div>
        <h2>Episode {this.props.episode}:</h2>
        <h3>{this.props.title}</h3>
      </div>
      <div>
        <img src={images[this.props.episode - 1]} alt='movie cover art'/>
      </div>
      <div>
        <p>Released:</p>
        <p>{this.props.releaseDate}</p>
      </div>
      <button className='characters-button' data-category={(this.props.episode - 1)} onClick={this.findCurrentMovie}>View Characters</button>
    </article>
    )}
}

export default MovieCard