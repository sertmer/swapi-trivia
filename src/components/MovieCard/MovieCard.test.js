import React from 'react'
import MovieCard from './MovieCard'
import { shallow } from 'enzyme'

describe('MovieCard', () => {
  let wrapper
  let mockUpdateState = jest.fn()
  let mockProps = {
    episode: 1,
    title: 'spaceballs',
    releaseDate: '12-12-12',
    updateState: mockUpdateState
  }
  beforeEach(() => {
    wrapper = shallow(
    <MovieCard 
      episode={1}
      title='title'
      releaseDate='12-12-12'
      updateState={jest.fn}
    />)
  })

  it('should match the snapshot', () => {   
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state when findCurrentMovie is called', () => {
    const mockEvent = {target: {dataset: {
      category: 1
    }}}
    wrapper.instance().findCurrentMovie(mockEvent)
    expect(wrapper.state('currentMovie')).toEqual(1)
  })

  it('should call updateState when findCurrentMovie is invoked', () => {
    const mockEvent = {target: {dataset: {
      category: 1
    }}}
    wrapper.instance().findCurrentMovie(mockEvent)
    expect(wrapper.instance().props.updateState).toHaveBeenCalledWith({currentMovie: 1})
  })
})