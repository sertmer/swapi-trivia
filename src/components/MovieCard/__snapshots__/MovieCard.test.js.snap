// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<article
  className="MovieCard"
>
  <div>
    <h2>
      Episode 
      :
    </h2>
    <h3 />
  </div>
  <div>
    <img
      alt="movie cover art"
    />
  </div>
  <div>
    <p>
      Released:
    </p>
    <p />
  </div>
  <Link
    to="/movies/:NaN"
  >
    <button
      className="characters-button"
      data-category={NaN}
      onClick={[Function]}
    >
      View Characters
    </button>
  </Link>
</article>
`;
